src/qibo/quantum_info/metrics.py

```diff
@@ -845,6 +845,27 @@ def frame_potential(
     return potential / samples**2
 
 
+def quantum_fisher_information_matrix(
+    circuit, parameters=None, return_complex: bool = True, backend=None
+):
+    backend = _check_backend(backend)
+
+    if parameters is None:
+        parameters = circuit.get_parameters()
+        parameters = backend.cast(parameters, dtype=float)
+
+    jacobian = backend.calculate_jacobian_matrix(circuit, parameters, return_complex)
+
+    original_parameters = circuit.get_parameters()
+    circuit.set_parameters(parameters)
+    state = backend.execute_circuit(circuit).state()
+    circuit.set_parameters(original_parameters)
```

Comment:
Instead of doing this, you may just decide to copy the circuit in case the `parameters` argument is specified
```suggestion
    if parameters is not None:
        circuit = copy.deepcopy(circuit)
        circuit.set_parameters(parameters)
    else:
        parameters = backend.cast(circuit.get_parameters(), dtype=float)

    jacobian = backend.calculate_jacobian_matrix(circuit, parameters, return_complex)
    state = backend.execute_circuit(circuit).state()
```

To me, it seems conceptually simpler.